### PR TITLE
Updating schema fields and sdk version

### PR DESCRIPTION
**Pull request by mchavez**
- Updating schema fields and sdk version

### CODE REVIEW

File: cmd/baton-cloudflare/config.go [hunk 1]
```diff
@@ -4,41 +4,32 @@ import (
...
-// config defines the external configuration required for the connector to run.
-type config struct {
-	cli.BaseConfig `mapstructure:",squash"` // Puts the base config options in the same place as the connector options
-
-	ApiKey    string `mapstructure:"api-key"`
-	ApiToken  string `mapstructure:"api-token"`
-	AccountId string `mapstructure:"account-id"`
-	EmailId   string `mapstructure:"email-id"`
-}
+var (
```
Comment (mgaeta) [inline cmd/baton-cloudflare/config.go:12]:
```go
var (
	ApiKeyField = field.StringField(
		"api-key",
		field.WithDescription("The api key for the Cloudflare account."),
	)
	ApiTokenField = field.StringField(
		"api-token",
		field.WithDescription("The api token for the Cloudflare account."),
	)
	AccountIdField = field.StringField(
		"account-id",
		field.WithDescription("The account id for the Cloudflare account."),
		field.WithRequired(true),
	)
	EmailIdField = field.StringField(
		"email-id",
		field.WithDescription("The email id for the Cloudflare account."),
	)
	configurationFields = []field.SchemaField{
		ApiKeyField,
		ApiTokenField,
		AccountIdField,
		EmailIdField,
	}
	fieldRelationships = []field.SchemaFieldRelationship{
		field.FieldsAtLeastOneUsed(
			ApiTokenField,
			ApiKeyField,
		),
		field.FieldsDependentOn(
			[]field.SchemaField{
				ApiKeyField,
			},
			[]field.SchemaField{
				EmailIdField,
			},
		),
	}
)
```

**Review by mgaeta — CHANGES_REQUESTED** [review]

File: pkg/connector/connector.go [hunk 2]
```diff
@@ -9,39 +9,51 @@ import (
...
 
-func New(ctx context.Context, config Config) (*Cloudflare, error) {
+const (
+	ApiKey    = "api-key"
+	ApiToken  = "api-token"
+	AccountId = "account-id"
+	EmailId   = "email-id"
+)
+
+func New(ctx context.Context, cfg *viper.Viper) (*Cloudflare, error) {
```
Comment (mgaeta) [inline pkg/connector/connector.go:24]: Keep the original method signature.

Comment (mchavez) [reply pkg/connector/connector.go]: Fixed

File: cmd/baton-cloudflare/config.go (hunk 1 above)
Comment (mchavez) [reply cmd/baton-cloudflare/config.go]: Fixed

File: cmd/baton-cloudflare/main.go [hunk 3]
```diff
@@ -6,40 +6,62 @@ import (
...
+	"github.com/spf13/viper"
 	"go.uber.org/zap"
 )
 
-var version = "dev"
+var (
+	apiKeyField         = field.StringField(connector.ApiKey, field.WithDescription("The api key for the Cloudflare account."))
+	apiTokenField       = field.StringField(connector.ApiToken, field.WithDescription("The api token for the Cloudflare account."))
+	accountIdField      = field.StringField(connector.AccountId, field.WithRequired(true), field.WithDescription("The account id for the Cloudflare account."))
+	emailIdField        = field.StringField(connector.EmailId, field.WithRequired(true), field.WithDescription("The email id for the Cloudflare account."))
```
Comment (mgaeta) [inline cmd/baton-cloudflare/main.go:22]: `emailIdField` should not be required.

```go
EmailIdField = field.StringField(
  "email-id",
  field.WithDescription("The email id for the Cloudflare account."),
)
```

**Review by mgaeta — CHANGES_REQUESTED** [review]

File: cmd/baton-cloudflare/main.go (hunk 3 above)
Comment (mchavez) [reply cmd/baton-cloudflare/main.go]: Fixed

**Review by mgaeta — APPROVED** [review]

File: cmd/baton-cloudflare/main.go [hunk 4]
```diff
@@ -6,40 +6,66 @@ import (
...
 
-func getConnector(ctx context.Context, cfg *config) (types.ConnectorServer, error) {
+func getConnector(ctx context.Context, cfg *viper.Viper) (types.ConnectorServer, error) {
 	l := ctxzap.Extract(ctx)
 	config := connector.Config{
-		AccountId: cfg.AccountId,
-		ApiToken:  cfg.ApiToken,
-		EmailId:   cfg.EmailId,
-		ApiKey:    cfg.ApiKey,
+		AccountId: cfg.GetString(accountId),
```
Comment (ggreer) [inline cmd/baton-cloudflare/main.go:65]:
```suggestion
		AccountId: cfg.GetString(accountIdField.FieldName),
```
You can use `.FieldName` for all of these, allowing you to put the actual names in the calls to `field.StringField(...)` and delete the consts on lines 33-40.

**Comment by ggreer** [comment]
Weird that CI is failing. This branch fetches the same resources as master, but I guess something changed in cloudflare so that we're not seeing the grant anymore. I think clean up the extra variables and this is good to go.

**Comment by mgaeta** [comment]
@mchavez Can I close this PR?

**Pull request merged**